internal/status/status.go

```diff
@@ -0,0 +1,61 @@
+package status
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/kverrors"
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+	"github.com/ViaQ/loki-operator/internal/external/k8s"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// Refresh executes an aggregate update of the LokiStack Status struct, i.e.
+// - It recreates the Status.Components pod status map per component.
+// - It sets the appropriate Status.Condition to true that matches the pod status maps.
+func Refresh(ctx context.Context, k k8s.Client, req ctrl.Request) error {
+	if err := SetComponentsStatus(ctx, k, req); err != nil {
```

Comment:
side question, why is this not using the reference of req? wouldn't that save us a lot of needing to do deep copies of the status and then updating it? the req object would always reflect the most up-to-date and accurate state then...

Reply:
AFAICS, the `DeepCopy` calls are all removed. The whole status pkg is designed to be self-sufficient in terms of requesting the `LokiStack` anew on public functions, e.g. `SetComponentsStatus`. Passing a reference to the LokiStack here makes error handling of situation like having an old version of `LokiStack` an issue for the calling site.